From 2dac940a7ae763f67c365d9c769357a2b7b579c5 Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Wed, 17 Nov 2021 19:20:53 +0100
Subject: [PATCH] DRBD transport: fixed a BSOD introduced with last commit.

Now we have several pages of PAGE_SIZE maximum size.
---
 drbd/drbd_transport_tcp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drbd/drbd_transport_tcp.c b/drbd/drbd_transport_tcp.c
index d7ea1d1..26fbf22 100644
--- a/drbd/drbd_transport_tcp.c
+++ b/drbd/drbd_transport_tcp.c
@@ -350,7 +350,7 @@ static int dtt_recv_pages(struct drbd_transport *transport, struct drbd_page_cha
 		return -ENOMEM;
 
 	page_chain_for_each(page) {
-		size_t len = size;
+		size_t len = min_t(int, size, PAGE_SIZE);
 		void *data = kmap(page);
 		err = dtt_recv_short(socket, data, len, 0);
 		kunmap(page);
-- 
2.17.0

